evap/contributor/tests/test_forms.py

```diff
@@ -63,6 +63,115 @@ def test_staff_only(self):
+            "contributions-0-DELETE": "",
+        }
+
+        formset = InlineContributionFormset(data, instance=course, can_edit_responsibles=False, form_kwargs={'course': course})
+
+        self.assertFalse(formset.is_valid())
+
+    def test_editors_cannot_delete_responsibles(self):
+        course = mommy.make(Course)
+        user = mommy.make(UserProfile)
+        questionnaire = mommy.make(Questionnaire, is_for_contributors=True)
+        contribution = mommy.make(Contribution, course=course, contributor=user, responsible=True,
+                                  can_edit=True, comment_visibility=Contribution.ALL_COMMENTS)
+        InlineContributionFormset = inlineformset_factory(Course, Contribution, formset=ContributionFormSet, form=EditorContributionForm, extra=1)
+
+        data = {
+            "contributions-TOTAL_FORMS": "1",
+            "contributions-INITIAL_FORMS": "1",
+            "contributions-MIN_NUM_FORMS": "0",
+            "contributions-MAX_NUM_FORMS": "1000",
+            "contributions-0-course": "{}".format(course.pk),
+            "contributions-0-order": "1",
+            "contributions-0-id": "{}".format(contribution.pk),
+            "contributions-0-contributor": "{}".format(user.pk),
+            "contributions-0-questionnaires": "{}".format(questionnaire.pk),
+            "contributions-0-responsibility": "RESPONSBILE",
+            "contributions-0-comment_visibility": "OWN",
+            "contributions-0-label": "",
+            "contributions-0-DELETE": "1",
+        }
+
+        formset = InlineContributionFormset(data, instance=course, can_edit_responsibles=False, form_kwargs={'course': course})
+
+        self.assertFalse(formset.is_valid())
+
+    def test_editors_cannot_add_responsibles(self):
+        course = mommy.make(Course)
+        user = mommy.make(UserProfile)
+        questionnaire = mommy.make(Questionnaire, is_for_contributors=True)
+        InlineContributionFormset = inlineformset_factory(Course, Contribution, formset=ContributionFormSet, form=EditorContributionForm, extra=1)
+
+        data = {
+            "contributions-TOTAL_FORMS": "1",
+            "contributions-INITIAL_FORMS": "0",
+            "contributions-MIN_NUM_FORMS": "0",
+            "contributions-MAX_NUM_FORMS": "1000",
+            "contributions-0-course": "{}".format(course.pk),
+            "contributions-0-order": "1",
+            "contributions-0-id": "",
+            "contributions-0-contributor": "{}".format(user.pk),
+            "contributions-0-questionnaires": "{}".format(questionnaire.pk),
+            "contributions-0-responsibility": "RESPONSBILE",
+            "contributions-0-comment_visibility": "OWN",
+            "contributions-0-label": "",
+            "contributions-0-DELETE": "1",
+        }
+
+        formset = InlineContributionFormset(data, instance=course, can_edit_responsibles=False, form_kwargs={'course': course})
+
+        self.assertFalse(formset.is_valid())
```

Comment:
i would prefer to first have some valid data, asserting that the form is valid, and then changing the necessary data to make the data invalid, and then asserting that the form is invalid.

there might be all kinds of other reasons for the data to be invalid, and right now it's not guaranteed that the test actually tests what you want it to test.